ntripclient.c

```diff
@@ -1636,24 +1696,60 @@ int main(int argc, char **argv)
                   starttime = time(0);
                   lastout = starttime;
                 }
-                if(args.serdevice && !stop)
+
+                // get subsequent NMEA GGA sentence from GGA source and
+                // send it to NTRIP caster.  Source can be either serial device
+                // or IPC pipe
+                if(args.nmea && !stop)
                 {
                   int doloop = 1;
                   while(doloop && !stop)
                   {
-                    int i = SerialRead(&sx, buf, 200);
+                    int i = 0;
+                    // get NMEA GGA sentence from serial device or IPC pipe
+                    if (args.serdevice) {
+                      i = SerialRead(&sx, buf, 200);
+                    } else {
+
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "requesting IPC receive\n");
+#endif // NTRIPCLIENT_DEBUG_LOG
+
+                      i = fread(buf, 1, 200, stdin);
+
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "IPC receive %d bytes\n", i);
+                      int k = 0;
+                      for (k = 0;  k < i; k++) {
```

Comment:
Can the scope of `k` be reduced to inside the for loop? e.g. declaration `for (int k = 0; ...)`

Reply:
Done!